res/controllers/Traktor-Kontrol-S3-hid-scripts.js

```diff
@@ -60,2149 +96,2160 @@ TraktorS3.ChannelColors = {
+
+        this.registerInputScaler(messageLong, "[EqualizerRack1_[Channel2]_Effect1]", "parameter3", 0x2B, 0xFFFF, this.parameterHandler);
+        this.registerInputScaler(messageLong, "[EqualizerRack1_[Channel2]_Effect1]", "parameter2", 0x2D, 0xFFFF, this.parameterHandler);
+        this.registerInputScaler(messageLong, "[EqualizerRack1_[Channel2]_Effect1]", "parameter1", 0x2F, 0xFFFF, this.parameterHandler);
+
+        this.registerInputScaler(messageLong, "[EqualizerRack1_[Channel3]_Effect1]", "parameter3", 0x1F, 0xFFFF, this.parameterHandler);
+        this.registerInputScaler(messageLong, "[EqualizerRack1_[Channel3]_Effect1]", "parameter2", 0x21, 0xFFFF, this.parameterHandler);
+        this.registerInputScaler(messageLong, "[EqualizerRack1_[Channel3]_Effect1]", "parameter1", 0x23, 0xFFFF, this.parameterHandler);
+
+        this.registerInputScaler(messageLong, "[EqualizerRack1_[Channel4]_Effect1]", "parameter3", 0x31, 0xFFFF, this.parameterHandler);
+        this.registerInputScaler(messageLong, "[EqualizerRack1_[Channel4]_Effect1]", "parameter2", 0x33, 0xFFFF, this.parameterHandler);
+        this.registerInputScaler(messageLong, "[EqualizerRack1_[Channel4]_Effect1]", "parameter1", 0x35, 0xFFFF, this.parameterHandler);
+
+        this.registerInputScaler(messageLong, "[Master]", "crossfader", 0x0B, 0xFFFF, this.parameterHandler);
+        this.registerInputScaler(messageLong, "[Master]", "gain", 0x17, 0xFFFF, TraktorS3.Controller.prototype.masterGainHandler.bind(this));
+        this.registerInputScaler(messageLong, "[Master]", "headMix", 0x1D, 0xFFFF, this.parameterHandler);
+        this.registerInputScaler(messageLong, "[Master]", "headGain", 0x1B, 0xFFFF, this.parameterHandler);
+
+        this.hid.registerInputPacket(messageLong);
+
+        for (const ch in this.Channels) {
+            const chanob = this.Channels[ch];
+            engine.makeConnection(ch, "playposition",
+                TraktorS3.Channel.prototype.playpositionChanged.bind(chanob));
+            engine.connectControl(ch, "track_loaded",
+                TraktorS3.Channel.prototype.trackLoadedHandler.bind(chanob));
+            engine.connectControl(ch, "end_of_track",
+                TraktorS3.Channel.prototype.endOfTrackHandler.bind(chanob));
+        }
 
-    // Represents the first-pressed deck switch button, used for tracking deck clones.
-    this.deckSwitchPressed = "";
+        // Query the current values from the controller and set them. The packet
+        // parser ignores the first time a value is set, so we'll need to set it
+        // with different values once. Report 2 contains the state of the mixer
+        // controls.
+        const report2Values = new Uint8Array(controller.getInputReport(2));
+        TraktorS3.incomingData([2, ...Array.from(report2Values.map(x => ~x))]);
+        TraktorS3.incomingData([2, ...Array.from(report2Values)]);
```

Comment:
```suggestion
        TraktorS3.incomingData(new Uint8Array([2, ...Uint8Array.from(report2Values.map(x => ~x))]));
        TraktorS3.incomingData(new Uint8Array([2, ...Uint8Array.from(report2Values)]));
```
When incomingData is called from C++, the argument is an array of type Uint8Array. The same type should be used here too.

Reply:
Done.